packages/frontpage/lib/data/atproto/identity.ts

```diff
@@ -2,35 +2,46 @@ import { cache } from "react";
 import { DID, getDidDoc, isDid, parseDid } from "./did";
 import { unstable_cache } from "next/cache";
 import { z } from "zod";
-
-export const getVerifiedDid = cache(async (handle: string) => {
-  const [dnsDid, httpDid] = await Promise.all([
-    getAtprotoDidFromDns(handle).catch((_) => {
-      return null;
-    }),
-    getAtprotoFromHttps(handle).catch((_) => {
-      return null;
-    }),
-  ]);
-
-  if (dnsDid && httpDid && dnsDid !== httpDid) {
-    return null;
-  }
-
-  const did = dnsDid ?? (httpDid ? parseDid(httpDid) : null);
-  if (!did) {
-    return null;
-  }
-
-  const plcDoc = await getDidDoc(did);
-  const plcHandle = plcDoc.alsoKnownAs
-    .find((handle) => handle.startsWith("at://"))
-    ?.replace("at://", "");
-
-  if (!plcHandle) return null;
-
-  return plcHandle.toLowerCase() === handle.toLowerCase() ? did : null;
-});
+import { startSpan } from "@sentry/nextjs";
+
+export const getVerifiedDid = cache((handle: string) =>
+  startSpan({ name: "getVerifiedDid" }, () =>
+    unstable_cache(
+      async () => {
+        const [dnsDid, httpDid] = await Promise.all([
+          getAtprotoDidFromDns(handle).catch((_) => {
+            return null;
+          }),
+          getAtprotoFromHttps(handle).catch((_) => {
+            return null;
+          }),
+        ]);
+
+        if (dnsDid && httpDid && dnsDid !== httpDid) {
+          return null;
+        }
+
+        const did = dnsDid ?? (httpDid ? parseDid(httpDid) : null);
+        if (!did) {
+          return null;
+        }
+
+        const plcDoc = await getDidDoc(did);
```

Comment:
There are still a bunch of extra cache reads coming from calling `getDidDoc` inside `getVerifiedHandle`...

I think the best thing to do would be to move to a `resolveIdentity` function like we do in atproto-browser - a function that accepts a handle or a DID and returns the did along with the did doc.